src/utils/xhr.js

```diff
@@ -0,0 +1,30 @@
+import {
+  get,
+  forEach,
+} from 'lodash';
+
+export const xhrAddHeaders = (xhr, headers) => {
+  forEach(headers, (value, key) => {
+    xhr.setRequestHeader(key, value);
+  });
+};
+
+export const getXHRResponse = response => {
+  const contentType = response.headers.get('Content-Type') || '';
+
+  if (!contentType.startsWith('application/json')) {
```

Comment:
Consider rewriting without - `!`
```suggestion
  if (contentType.startsWith('application/json')) {
    return response.json();
  }

  return response;
```

And is it really needed to use `startsWith`? Why not to just write `contentType === 'application/json'`